libraries/AP_Compass/AP_Compass_AK09916.cpp

```diff
@@ -58,7 +58,7 @@ extern const AP_HAL::HAL &hal;
 
 extern const AP_HAL::HAL &hal;
 
-struct PACKED sample_regs {
+struct PACKED sample_regs_AK09916 {
```

Comment:
Why not namespace by putting in `private` in the class?

Reply:
I tried but the definition is needed in e.g. `AP_AK09916_BusDriver_Auxiliary::start_measurements()` and I'm not sure how to work out the visibility there.

Reply:
Resolved to put them in the class's public namespace where necessary. They were essentially public before so it shouldn't be a major concern.